platform/macos/src/MGLMapView.mm

```diff
@@ -1275,6 +1275,30 @@ - (MGLMapCamera *)cameraThatFitsCoordinateBounds:(MGLCoordinateBounds)bounds edg
     return [self cameraForCameraOptions:cameraOptions];
 }
 
+- (MGLMapCamera *)camera:(MGLMapCamera *)camera fittingCoordinateBounds:(MGLCoordinateBounds)bounds edgePadding:(NSEdgeInsets)insets
+{
+    mbgl::EdgeInsets padding = MGLEdgeInsetsFromNSEdgeInsets(insets);
+    padding += MGLEdgeInsetsFromNSEdgeInsets(self.contentInsets);
+    
+    CGFloat pitch = camera.pitch < 0 ? _mbglMap->getPitch() : camera.pitch;
+    CLLocationDirection direction = camera.heading < 0 ? _mbglMap->getBearing() : camera.heading;
+    
+    mbgl::CameraOptions cameraOptions = _mbglMap->cameraForLatLngBounds(MGLLatLngBoundsFromCoordinateBounds(bounds), padding, direction, pitch);
+    return [self cameraForCameraOptions:cameraOptions];
+}
+
+- (MGLMapCamera *)camera:(MGLMapCamera *)camera fittingShape:(MGLShape *)shape edgePadding:(NSEdgeInsets)insets {
+    mbgl::EdgeInsets padding = MGLEdgeInsetsFromNSEdgeInsets(insets);
+    padding += MGLEdgeInsetsFromNSEdgeInsets(self.contentInsets);
+    
+    CGFloat pitch = camera.pitch < 0 ? _mbglMap->getPitch() : camera.pitch;
+    CLLocationDirection direction = camera.heading < 0 ? _mbglMap->getBearing() : camera.heading;
```

Comment:
When possible, use existing property getters to ensure that normalization is applied consistently:

```objc
MGLMapCamera *currentCamera = self.camera;
CGFloat pitch = camera.pitch < 0 ? currentCamera.pitch : camera.pitch;
CLLocationDirection direction = camera.heading < 0 ? currentCamera.heading : camera.heading;
```

(There’s also an `MGLMapView.direction`, but we have to call `-camera` anyways to get the current pitch.)

×4